### PR TITLE
185032522 outgoing referral jobs

### PR DESCRIPTION
**Pull request by ttoomey**
marked as draft pending resolution of issues in remote api

* live external referral tests
* oauth client: support patch http request, normalize urls

### CODE REVIEW

**Review by eanders — APPROVED** [review]
Looks good to me.  I'll defer to @gigxz or @yakloinsteak for the details

**Comment by eanders** [comment]
Oh, for some reason I thought the review was assigned to me.  Feel free to ignore since this is still a draft.

**Comment by gigxz** [comment]
Yep, let me know when you're ready for review @ttoomey . I did test this branch locally because they had mentioned the external issues were resolved (no dice, though) and it looks good to me. I don't think we necessarily need to wait to merge until the external issues are resolved.

File: drivers/hmis_external_apis/spec/models/hmis_external_apis/link_spec.rb [hunk 1]
```diff
@@ -5,38 +5,103 @@
...
+        'unitTypeID' => unit_type_id,
+        'estimatedDate' => format_date(now + 1.day),
+        'requestorName' => 'greenriver',
+        'requestorPhoneNumber' => '8028675309',
+        'requestorEmail' => 'test@greenriver.com',
+        'requestedBy' => requested_by,
+      }
+
+      result = subject.post('Referral/ReferralRequest', payload)
+      # {"message"=>"A server error occurred.", "detail"=>"ORA-00942: table or view does not exist", "errorReferenceId"=>"58e06257-3128-43d0-96fe-03cc6af4d8d6"}
```
Comment (gigxz) [inline drivers/hmis_external_apis/spec/models/hmis_external_apis/link_spec.rb]: Fyi @ttoomey this one and voiding a referral request are working now. The other two both still have outstanding issues that have been communicated:

```
Unit/Capacity
Request: "{\"programID\":1008,\"unitTypeID\":20,\"availableUnits\":3,\"requestedBy\":\"[test@greenriver.com](mailto:test@greenriver.com)\"}"
Response: "{\"message\":\"No Resource found with the values provided. \",\"detail\":\"Invalid Parameters\"}"

Referral/PostingStatus
Request: "{\"postingId\":3777,\"postingStatusId\":18,\"deniedReasonId\":1,\"referralResultId\":158,\"statusNote\":\"test\",\"contactDate\":\"2023-05-02T17:41:49+00:00\",\"requestedBy\":\"[test@greenriver.com](mailto:test@greenriver.com)\"}"
Response: "{\"message\":\"Posting Status Id is not valid. \",\"detail\":\"Invalid Parameters.\"}"
```

**Comment by ttoomey** [comment]
> Yep, let me know when you're ready for review @ttoomey . I did test this branch locally because they had mentioned the external issues were resolved (no dice, though) and it looks good to me. I don't think we necessarily need to wait to merge until the external issues are resolved.

@gigxz Thanks - I think this is ready for review. I'm fine with merging this once you review it; I can pick it up again when the external issues are resolved. Thanks!

**Review by gigxz — APPROVED** [review]

**Comment by gigxz** [comment]
Great, looks good to me to merge when tests pass. There was one oauth test failing @ttoomey .

**Pull request merged**